xapian-applications/omega/omindextest

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+# omindextest: Test omindex
+#
+# Copyright (C) 2019 Bruno Baruffaldi
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+# USA
+
+DIR=$(pwd)
+TEST_FILES="$DIR/testfiles/"
+TEST_DB="$DIR/testdatabase"
```

Comment:
This isn't right when building outside the source tree (which we aim to support).

Assuming `/testfiles/` is static data (i.e. it isn't generated as part of the build) than the correct place to look is `$srcdir` which will be set when `omindextest` is run via `make check` but not if `omindextest` is run by hand.

But the `omindextest` script is also not generated so should also be in the source tree, so we can use the directory part of `$0` to give us the path to the source directory, e.g.:

```
srcdir=`echo "$0"|sed 's!/*[^/]*$!!'`
```

So if you run `./omindextest` then that will set srcdir to `.` which is what we want.

If you're building in a `BUILD` subdirectory of the source tree, you would run `../omindextest` from there and then srcdir is set to `..`.  And a full path also works.

However `TEST_DB` should go in the build directory (because the source directory might be shared with other build directories, or might even be read-only).  The build directory is just the current directory, and there's no need for the database path to be an absolute path, so there's no need to run `pwd` - just `TESTDB=testdatabase` should work.

Reply:
Okay, I am going to change it.

Reply:
Okay, I am going to change it.